File: 60. Permutation Sequence.cpp
// timeout

class Solution {
private:
    vector <string> res;
public:
    string getPermutation(int n, int k) {
        vector <bool> visited (n, false);
        DFS("", n, k, visited);
        return res[k-1];
    }
    
    void DFS(string path, int n, int k, vector <bool> visited)
    {
        if (path.length() == n)
        {
            res.push_back(path);
            return;
        }
        if (res.size() < k)
        {
            for (int i = 1; i <= n; i++)
            {
                if (!visited[i-1])
                {
                    visited[i-1] = true;
                    DFS(path + to_string(i), n, k, visited);
                    visited[i-1] = false;
                }
            }
        }
    }
};

File: 60. Permutation Sequence.py
# timeout

class Solution(object):
    def getPermutation(self, n, k):
        """
        :type n: int
        :type k: int
        :rtype: str
        """
        self.res = []
        visited = [False for i in xrange(n)]
        self.DFS("", n, k, visited)
        return self.res[k-1]
        
    def DFS(self, path, n, k, visited):
        if len(path) == n:
            self.res += [path]
            return
        if len(self.res) < k:
            for i in xrange(1, n+1):
                if visited[i-1] == False:
                    visited[i-1] = True
                    self.DFS(path + str(i), n, k, visited)
                    visited[i-1] = False
